File: app/src/main/java/com/test/shoppingapp/shoppinglist/ShoppingListViewModel.java
package com.test.shoppingapp.shoppinglist;


import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import com.test.networkmodule.response.ShopListingResponse;
import com.test.shoppingapp.shoppinglist.repository.ShoppingRepository;

import java.util.List;

import javax.inject.Inject;

public class ShoppingListViewModel extends ViewModel {

    private static final int ALL = 0;
    private static final int MEN = 1;
    private static final int WOMEN = 2;

    private final ShoppingRepository shoppingRepository;

    @Inject
    public ShoppingListViewModel(ShoppingRepository shoppingRepository) {
        this.shoppingRepository = shoppingRepository;
    }

    public void findListFromIndex(int position) {
        switch (position) {
            case ALL:
                shoppingRepository.getShoppingListForAll();
                break;

            case MEN:
                shoppingRepository.getShoppingListForMen();
                break;

            case WOMEN:
                shoppingRepository.getShoppingListForWomen();
                break;
        }
    }

    public LiveData<List<ShopListingResponse>> getLiveDataForShoppingList(int position) {
        switch (position) {
            case ALL:
                return shoppingRepository.getAllShoppingList();

            case MEN:
                return shoppingRepository.getMensShoppingList();

            case WOMEN:
                return shoppingRepository.getWomensShoppingList();
        }
        return null;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        shoppingRepository.onCleared();
    }
}

File: app/src/main/java/com/test/shoppingapp/shoppinglist/ShoppingListViewModel.kt
package com.test.shoppingapp.shoppinglist


import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel

import com.test.networkmodule.response.ShopListingResponse
import com.test.shoppingapp.shoppinglist.repository.ShoppingRepository

import javax.inject.Inject

class ShoppingListViewModel @Inject
constructor(private val shoppingRepository: ShoppingRepository) : ViewModel() {

    fun findListFromIndex(position: Int) {
        when (position) {
            ALL -> shoppingRepository.getShoppingListForAll()

            MEN -> shoppingRepository.getShoppingListForMen()

            WOMEN -> shoppingRepository.getShoppingListForWomen()
        }
    }

    fun getLiveDataForShoppingList(position: Int): LiveData<List<ShopListingResponse>>? {
        when (position) {
            ALL -> return shoppingRepository.getAllShoppingList()

            MEN -> return shoppingRepository.getMensShoppingList()

            WOMEN -> return shoppingRepository.getWomensShoppingList()
        }
        return null
    }

    fun getErrorData(): LiveData<String> {
        return shoppingRepository.errorData
    }

    override fun onCleared() {
        super.onCleared()
        shoppingRepository.onCleared()
    }

    companion object {
        private const val ALL = 0
        private const val MEN = 1
        private const val WOMEN = 2
    }
}
